### PR TITLE
Change PodSecurityPolicy apiVersion

### PR DESCRIPTION
**Pull request by alxsap**
old one is deprecated

### CODE REVIEW

**Comment by anbrsap** [comment]
This will fail a Helm upgrade of an existing installation. Helm doesn't know that `podsecuritypolicy.extensions` and `podsecuritypolicy.policy` are the same. Hence it doesn't update the existing resource but tries to create the new one before deleting the old one. The creation fails because the K8s API Server detects that the resource exists already.

To prevent this, you may either require users to manually delete the old policy resource, or you rename the resource.

**Comment by alxsap** [comment]
I would vote for choosing a different name.
What happens with the old PSP once the cluster is upgraded? I assume it simply cannot be loaded, but since it is not referenced anymore this should not be a problem?

**Comment by anbrsap** [comment]
> I would vote for choosing a different name.
> What happens with the old PSP once the cluster is upgraded? I assume it simply cannot be loaded, but since it is not referenced anymore this should not be a problem?

Helm will delete it, because from its perspective it is a resource that was present in the previous release and is no longer present in the current release.

File: charts/steward/templates/podsecuritypolicy-run.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: extensions/v1beta1
+apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
   name: steward-run
```
Comment (alxsap) [inline charts/steward/templates/podsecuritypolicy-run.yaml]: For sorting reasons on Gardener we prefixed `00-` to ensure our PSP is used.
```suggestion
  name: 00-steward-run
```

**Comment by anbrsap** [comment]
> 
> 
> > I would vote for choosing a different name.
> > What happens with the old PSP once the cluster is upgraded? I assume it simply cannot be loaded, but since it is not referenced anymore this should not be a problem?
> 
> Helm will delete it, because from its perspective it is a resource that was present in the previous release and is no longer present in the current release.

Or did you mean a Kubernetes upgrade to v1.16 or higher? Any existing `podsecuritypolicy.extensions` object is internally stored as `podsecuritypolicy.policy`, but served under both API groups (extensions, policy). With K8s v1.16 the existing objects are still there and functional, but you cannot create or read them using the `extensions` API group.

In order to make our Helm chart work on K8s v1.16+ we need to change the API group to `policy`, as proposed here. To avoid problems with Helm upgrade, we also need to rename the policy object.

**Pull request merged**